src/malco/post_process/compute_mrr.py

```diff
@@ -42,17 +59,34 @@ def compute_mrr(output_dir, prompt_dir, correct_answer_file) -> Path:
         # df['correct_term'] is an OMIM
         # call OAK and get OMIM IDs for df['term'] and see if df['correct_term'] is one of them
         # in the case of phenotypic series, if Mondo corresponds to grouping term, accept it
-        df['is_correct'] = df.apply(
-            lambda row: score_grounded_result(row['term'], row['correct_term']) > 0,
-            axis=1)
+                        
+        #df['is_correct'] = df.apply(
+        #       lambda row: score_grounded_result(row['term'], row['correct_term'], mondo) > 0,
+         #   axis=1)
 
         # Calculate reciprocal rank
+        # Make sure caching is used in the following by unwrapping explicitly
+        results = []
+        for idx, row in df.iterrows():
+            val = score_grounded_result(row['term'], row['correct_term'], mondo)
+            is_correct = val > 0
+            results.append(is_correct)
+
+        df['is_correct'] = results
+
         df["reciprocal_rank"] = df.apply(
             lambda row: 1 / row["rank"] if row["is_correct"] else 0, axis=1
         )
         # Calculate MRR for this file
         mrr = df.groupby("label")["reciprocal_rank"].max().mean()
         mrr_scores.append(mrr)
+        print('=' * 100)
+        print('score_grounded_result cache info:\n')
+        print(score_grounded_result.cache_info())
+        print('=' * 100)
+        print('omim_mappings cache info:\n')
+        print(omim_mappings.cache_info())
+        print('=' * 100)
 
```

Comment:
It would be nice to also score the raw ranks here so that we can do other analysis such as the percentage of cases in the top 3 or top 10 ranks -- which seems to be the standard for this kind of comparison

Reply:
Moved this to 
https://github.com/monarch-initiative/malco/issues/28